### PR TITLE
Improve Attack animations

### PR DESCRIPTION
**Pull request by neersighted**
Ported my own animations from Bay.

Also improved code for /atom/movable/do_attack_animation.

:cl: neersighted
tweak: Attack animations are now directional!
/:cl:

Demo

### CODE REVIEW

**Comment by Anonus** [comment]
You should check if this messes up on lying people on our code, because, unlike bay, we rotate humans when we want them to be sideways instead of generating a sideways icon like bay does.

**Comment by MrPerson** [comment]
Resolvable in 509, if it's an issue.

**Comment by Anonus** [comment]
True, so it's not that big of a concern anyway.

**Comment by WJohn** [comment]
:+1: Extra feedback now tells you who is doing what.

**Comment by duncathan** [comment]
:+1:

**Comment by PKPenguin321** [comment]
so from what i can tell, this basically makes it so the little transparent image floats from the attacker towards the victim instead of just appearing over the victim?

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -807,24 +796,44 @@ Sorry Giacom. Please don't be mad :(
 /mob/living/do_attack_animation(atom/A)
 	var/final_pixel_y = get_standard_pixel_y_offset(lying)
 	..(A, final_pixel_y)
-	floating = 0 // If we were without gravity, the bouncing animation got stopped, so we make sure to restart it in next life().
+	floating = 0 // If we were without gravity, the bouncing animation got stopped, so we make sure we restart the bouncing after the next movement.
```
Comment (phil235) [inline code/modules/mob/living/living.dm]: gravity floating is restarted in life() not after a movement, afaik.

**Comment by RemieRichards** [comment]
@PKPenguin321 it used to/currently appear slightly above a mob and slam down on them, this instead makes it move directionally towards the mob, rather than just a generic slam downwards.

:+1: muh code babbu ;_;

**Comment by MrStonedOne** [comment]
Ya, a before/after would be helpful for prs like this.

**Pull request merged**